File: Golang/InsertionSortList/InsertionSortList_test.go
package InsertionSortList

import (
	"testing"
	"reflect"
)

func TestInsertionSortList(t *testing.T) {

	if insertionSortList(nil) == nil {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(
		insertionSortList(&ListNode{
			Val: 1,
		}),
		&ListNode{
			Val: 1,
		}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(
		insertionSortList(&ListNode{
			Val: 1,
			Next: &ListNode{
				Val: 0,
			},
		}),
		&ListNode{
			Val: 0,
			Next: &ListNode{
				Val: 1,
			},
		}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/InsertionSortListTest.py
from unittest import TestCase

from InsertionSortList import InsertionSortList, ListNode


class TestInsertionSortList(TestCase):
    def test_insertionSortList(self):
        ist = InsertionSortList()

        self.assertIsNone(ist.insertionSortList(None))

        self.assertEqual(ist.insertionSortList(ListNode(0)), ListNode(0))

        node0 = ListNode(1)
        node0.next = ListNode(0)
        node1 = ListNode(0)
        node1.next = ListNode(1)
        self.assertEqual(ist.insertionSortList(node0), node1)
